download.js

```diff
@@ -14,6 +14,10 @@
 
 'use strict';
 
+// Proxy support for got via global-agent
+// $ export GLOBAL_AGENT_HTTP_PROXY=
+const gaBootstrap = require('global-agent').bootstrap;
```

Comment:
Can we load it only if there are PROXY env variables detected?

Reply:
I would trust it to do the right thing and not any logic of my own.
But if you really wanted to i think this would work the same way?
```suggestion
// or
// $ export GLOBAL_AGENT_ENVIRONMENT_VARIABLE_NAMESPACE=
// $ export HTTP_PROXY=
const globalAgentEnvVarNS = typeof process.env.GLOBAL_AGENT_ENVIRONMENT_VARIABLE_NAMESPACE === 'string' ? process.env.GLOBAL_AGENT_ENVIRONMENT_VARIABLE_NAMESPACE : 'GLOBAL_AGENT_';
const globalAgentEnvVarHTTPsProxy = process.env[globalAgentEnvVarNS + 'HTTPS_PROXY'] ?? process.env[globalAgentEnvVarNS + 'HTTP_PROXY'] ?? null;
if (typeof globalAgentEnvVarHTTPsProxy === 'string') {
  const globalAgentBootstrap = require('global-agent').bootstrap;
  globalAgentBootstrap();
}
```
and removing `gaBootstrap()` of course.

Reply:
> I would trust it to do the right thing and not add any logic of my own.

Sure, but can we confirm that this package doesn't touch native interfaces if it won't find proxy configuration?

Reply:
Sounds like it would do something to the interface...
> By default, global-agent overrides [agent property](https://nodejs.org/api/http.html#http_http_request_options_callback) of any HTTP request, even if agent property was explicitly set when constructing a HTTP request.

I can't really say. But i can fix up the PR as i'm quite happy with the added piece of code.

Reply:
Ok, so let's go with it